tests/integration/dynamic_ip_test.py

```diff
@@ -577,6 +578,117 @@ def test_ipv6_autoconf_only(dhcpcli_up):
     libnmstate.apply(desired_state)
 
 
+def test_static_ipv4_with_routes_swich_back_to_dhcp(dhcpcli_up):
+    desired_state = dhcpcli_up
+    dhcp_cli_desired_state = desired_state[INTERFACES][0]
+    dhcp_cli_desired_state[Interface.STATE] = InterfaceState.UP
+    dhcp_cli_desired_state[Interface.IPV4] = create_ipv4_state(
+        enabled=True, dhcp=False
+    )
+    dhcp_cli_desired_state[Interface.IPV4][InterfaceIPv4.ADDRESS] = [
+        {
+            InterfaceIPv4.ADDRESS_IP: IPV4_ADDRESS2,
+            InterfaceIPv4.ADDRESS_PREFIX_LENGTH: 24,
+        }
+    ]
+
+    desired_state[RT.KEY] = {
+        RT.CONFIG: [
+            {
+                RT.DESTINATION: IPV4_DEFAULT_GATEWAY,
+                RT.NEXT_HOP_ADDRESS: DHCP_SRV_IP4,
+                RT.NEXT_HOP_INTERFACE: DHCP_CLI_NIC,
+            },
+            {
+                RT.DESTINATION: '203.0.113.0/24',
+                RT.NEXT_HOP_ADDRESS: DHCP_SRV_IP4,
+                RT.NEXT_HOP_INTERFACE: DHCP_CLI_NIC,
+            },
+        ]
+    }
+
+    libnmstate.apply(desired_state)
+    assertlib.assert_state(desired_state)
+
+    desired_state = dhcpcli_up
+    dhcp_cli_desired_state = desired_state[INTERFACES][0]
+    dhcp_cli_desired_state[Interface.STATE] = InterfaceState.UP
+    dhcp_cli_desired_state[Interface.IPV4] = create_ipv4_state(
+        enabled=True, dhcp=True
+    )
+
+    libnmstate.apply(desired_state)
+    assertlib.assert_state(desired_state)
+
+    time.sleep(5)  # wait to get resolv.conf updated
+
+    current_config_routes = [
+        route
+        for route in libnmstate.show()[RT.KEY][RT.CONFIG]
+        if route[RT.NEXT_HOP_INTERFACE] == DHCP_CLI_NIC
+    ]
+
+    assert _has_ipv4_dhcp_nameserver()
+    assert _has_ipv4_dhcp_gateway()
+    assert _has_ipv4_classless_route()
+    assert not current_config_routes
+
+
+def test_static_ipv6_with_routes_swich_back_to_autoconf(dhcpcli_up):
```

Comment:
Since the tests are already too slow, is it worth to test the IPv4/IPv6 behavior differently?

Reply:
Merged into `test_static_ip_with_routes_switch_back_to_dynamic`.